packages/web/src/pages/search-page/search-results/TrackResults.tsx

```diff
@@ -84,8 +86,11 @@ export const TrackResults = (props: TrackResultsProps) => {
     [dispatch, searchParams]
   )
 
+  // Wait for useSearchAllResults to finish loading before fetching tracks
   const { data, hasNextPage, loadNextPage, isPlaying, play, pause, lineup } =
-    useSearchTrackResults(searchParams)
+    useSearchTrackResults(searchParams, {
+      enabled: !waitForAllResults // Only fetch if not waiting for all results
```

Comment:
interesting, my guess is this would only fetch when we are on the tracks tab anyway?